finagle-core/src/test/scala/com/twitter/finagle/util/LoadServiceTest.scala

```diff
@@ -94,6 +95,12 @@ class LoadServiceTest extends FunSuite with MockitoSugar {
     executor.shutdown()
   }
 
+  test("LoadService should ignore packages according to ignoredPaths GlobalFlag") {
```

Comment:
Alright, so after further investigation as to why it was not picking up classes as expected based on the ignoredPackages, i tried the new Classloader as Steve suggested but that also yielded the same result.

I, however, did stumble across some interesting information. Several classes like Resolver, Announcer, LoadServiceMaybeInterface and LoadServiceRandomInterface were actually being picked up from "META-INF/services/" and then the getName for those files contained their fully qualified names such as "com.twitter.finagle.util.LoadServiceRandomInterface". Therefore, the only way i was able to exclude it from the LoadService scan was to add "META-INF/services/" to the ignoredPaths via the GlobalFlag.

I am not sure if this would be very readable test unless i add some sort of explanation there but if you have any idea as to how i could have a more straightforward test, it would be most appreciated.

On a not so related note, i was wondering if you could tell me how i could use the GlobalFlag programmatically in my application. Setting it via command-line args isn't really possible for us since we do not really use this as a complete Twitter App. We directly start up the Thrift/ThriftMux clients and this LoadService is simply a by-product for the scan for Resolvers when we try to do a bind. It would be nice to know if there is some way to set GlobalFlags within the code (besides the let() method).